src/recon_buildblock/Parallelproj_projector/BackProjectorByBinParallelproj.cxx

```diff
@@ -159,7 +167,32 @@ get_output(DiscretisedDensity<3,float> &density) const
       else{
         num_lors_per_chunk = num_lors_per_chunk_floor;
       }
-
+      if (p.get_proj_data_info_sptr()->is_tof_data() == 1)
+        {
+          info("running the CUDA version of parallelproj, about to call function joseph3d_back_tof_sino_cuda", 2);
+
+          std::vector<float> mem_for_PP_back(num_lors_per_chunk * _helper->num_tof_bins);
+          const float* STIR_mem = p.get_const_data_ptr();
+
+          TOF_transpose(mem_for_PP_back, STIR_mem, _helper, offset);
```

Comment:
Should this be cached? My understanding is `mem_for_PP_back` is a "sorted" `STIR_mem` with `offset` taken into account.

Reply:
can't cache anything here. it's dependent on the data.